src/http_server/api/v1/metrics.c

```diff
@@ -34,6 +35,8 @@
 
 #define PROMETHEUS_HEADER "text/plain; version=0.0.4"
 
+#define sds_null_check(x) do { if (!x) goto error; } while (0)
```

Comment:
I don't have a lot of experience with macros- but this confuses me- why wrap the check in a `do while`?

Reply:
Its kind of best practice to be save with semicolon usage - https://stackoverflow.com/questions/154136/why-use-apparently-meaningless-do-while-and-if-else-statements-in-macros